test/storage/index_manager_test.cpp

```diff
@@ -0,0 +1,117 @@
+  auto table_entry = table_handle.GetTableEntry(txn0, "test_table");
+  EXPECT_NE(table_entry, nullptr);
+  std::string_view str = type::TransientValuePeeker::PeekVarChar(table_entry->GetColInRow(0));
+  EXPECT_EQ(str, "public");
+
+  str = type::TransientValuePeeker::PeekVarChar(table_entry->GetColInRow(1));
+  EXPECT_EQ(str, "test_table");
+
+  str = type::TransientValuePeeker::PeekVarChar(table_entry->GetColInRow(2));
+  EXPECT_EQ(str, "pg_default");
+
+  // Insert a few rows into the table
+  auto ptr = table_handle.GetTable(txn0, "test_table");
+  EXPECT_EQ(ptr, table);
+
+  for (int i = 0; i < 200; ++i) {
+    std::vector<type::TransientValue> row;
+    std::ostringstream stringStream;
+    stringStream << "name_" << i;
+    std::string name_string = stringStream.str();
+    stringStream << "address_" << i;
+    std::string address_string = stringStream.str();
+    row.emplace_back(type::TransientValueFactory::GetBoolean(i % 2 == 0));
+    row.emplace_back(type::TransientValueFactory::GetInteger(i));
+    row.emplace_back(type::TransientValueFactory::GetVarChar(name_string));
+    row.emplace_back(type::TransientValueFactory::GetVarChar(address_string));
+    ptr->InsertRow(txn0, row);
+  }
+
+  // Commit the setting transaction
+  txn_manager_->Commit(txn0, TestCallbacks::EmptyCallback, nullptr);
+
+  // Set up index attributes and key attributes
+  std::vector<std::string> index_attrs{"sex", "id", "name"};
+  std::vector<std::string> key_attrs{"id", "name"};
+
+  // Create the index
+  auto index_oid = index_manager_->CreateConcurrently(terrier_oid, ns_oid, table_oid, parser::IndexType::BWTREE, false,
+                                                      "test_index", index_attrs, key_attrs, txn_manager_, catalog_);
+
+  // Test whether the catalog has the corresponding information
+  auto txn1 = txn_manager_->BeginTransaction();
+  EXPECT_GT(!index_oid, 0);
+  auto index_handle = db_handle.GetIndexHandle(txn1, terrier_oid);
+  auto index_entry = index_handle.GetIndexEntry(txn1, index_oid);
+  EXPECT_NE(index_entry, nullptr);
+  auto ret = index_entry->GetIntegerColumn("indexrelid");
+  EXPECT_EQ(ret, !index_oid);
+  EXPECT_EQ(index_entry->GetIntegerColumn("indrelid"), !table_oid);
+  EXPECT_EQ(index_entry->GetBooleanColumn("indisready"), false);
+  EXPECT_EQ(index_entry->GetBooleanColumn("indisvalid"), true);
+  txn_manager_->Commit(txn1, TestCallbacks::EmptyCallback, nullptr);
+
+  // Drop the index and test the information does not exist
+  index_manager_->Drop(terrier_oid, ns_oid, table_oid, index_oid, "test_index", txn_manager_, catalog_);
+  auto txn2 = txn_manager_->BeginTransaction();
+  index_entry = index_handle.GetIndexEntry(txn2, index_oid);
+  EXPECT_EQ(index_entry, nullptr);
+  txn_manager_->Commit(txn2, TestCallbacks::EmptyCallback, nullptr);
+
+  delete table;
+  delete catalog_;
+  delete txn0;
+  delete txn1;
+  delete txn2;
+}
+}  // namespace terrier::storage::index
```

Comment:
You mentioned an edge case in your design document where a key could be inserted twice into the index. Therefore, there is supposed to be a test case on that in your unit test.

Reply:
That is actually the concurrent test. We will add it later.